src/app-env.ts

```diff
@@ -98,14 +98,19 @@ export class TestEnvironment<S extends TestServerApi = TestServerApi> {
     return tab;
   }
 
-  public async closeTab() {
-    // TODO: Call ensureNoErrors() on the tab
+  public async closeTab(targetId: string) {
+    if (!(targetId in this.targetIdToClientEnv)) {
+      throw new Error(`Can't close tab with targetId:{$targetId}. ` +
+                      'This targetId was not created within this TestEnvironment');
```

Comment:
nit: technically we don't know that it wasn't created, we just know it no longer exists.

Reply:
I'm assuming there is no other way to close a tab in our system. Just in case, I've made it more ambiguous with "This targetId is not associated with this TestEnvironment"